File: Blind75/Validate Binary Search Tree/Validate_Binary_Search_Tree.js
function binarySearchTreeValidate(root, low = null, high = null){
    if (!root) return true
    if(low !== null && low >= root.val || 
        high !== null && high <= root.val
    )
    return false
  
    let leftTree = binarySearchTreeValidate(root.left, low, root.val)
    let rightTree = binarySearchTreeValidate(root.right, root.val, high)

    return leftTree && rightTree
}

File: Blind75/Validate Binary Search Tree/Validate_Binary_Search_Tree.py
from typing import Optional

def isValidBST(root, low: Optional[int] = None, high: Optional[int] = None) -> bool:
    if root is None: return True
    if low is not None and low >= root.val or high is not None and high <= root.val: return False

    left_tree = isValidBST(root.left, low, root.val)
    right_tree = isValidBST(root.right, root.val, high)

    return left_tree and right_tree
